packages/eslint-plugin/src/rules/analytics-configuration.ts

```diff
@@ -213,6 +215,35 @@ export const analyticsConfiguration = OpenverseRule<Options, MessageIds>({
           if (m.type === "TSPropertySignature") validateCustomEvent(m)
         })
       },
+
+      IfStatement(node: TSESTree.IfStatement) {
+        if (!node.consequent || node.consequent.type !== 'BlockStatement') {
+          context.report({
+            node,
+            messageId: 'missingBraces',
+          });
+        }
+
+        if (node.alternate && node.alternate.type !== 'BlockStatement') {
+          context.report({
+            node,
+            messageId: 'missingBraces',
+          });
+        }
+      },
+
+      SwitchStatement(node: TSESTree.SwitchStatement) {
+        if (node.cases) {
+          for (const switchCase of node.cases) {
+            if (!switchCase.consequent || switchCase.consequent.length === 0) {
+              context.report({
+                node: switchCase,
+                messageId: 'missingBraces',
+              });
+            }
+          }
+        }
+      },
```

Comment:
Although I can't say for sure without extensively testing it locally, this rule, from a first glance, seems to me to be implemented correctly. However, this specific file `packages/eslint-plugin/src/rules/analytics-configuration.ts` is supposed to be for our rules specifically related to the analytics area. You should put this in a separate, new file under `packages/eslint-plugin/src/rules/`.

Reply:
Really we don't need this custom implementation at all. The ESLint unicorn plugin already has this rule, and we already use it. The unicorn implementation also has automatic fixes, which would solve some of the other issues in this PR as well caused by manually adding the braces.

Here's the unicorn rule: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/switch-case-braces.md

To enable it, add it to project's ESLint configuration here: https://github.com/WordPress/openverse/blob/8b9dabe011b133f7edfc84e59ddd58db57872d38/packages/eslint-plugin/src/configs/index.ts#L32-L36

```
"unicorn/switch-case-braces": ["error"],
```

It also won't incorrectly cause an error on fall-through cases like this implemention does here: https://github.com/WordPress/openverse/pull/3459/files#diff-b3a01ee15275f0b7ecd3aa87dec5fefd76e9f6d324a11042ce4c9bb3a4fb500eR428